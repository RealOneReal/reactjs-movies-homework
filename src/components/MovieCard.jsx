import {
  Card,
  Typography,
  CardMedia,
  CardContent,
  Avatar,
  Box,
} from "@mui/material";
import { PlayCircle } from "@mui/icons-material";
import { truncateString } from "../utils/stringUtils";

const MovieCard = (props) => {
  const { poster_path, genre_ids, vote_average, original_title, id } = props;
  const title = truncateString(original_title, 19);
  const genres = truncateString(genre_ids?.length && genre_ids.join(" "), 26);

  const handleClick = () => {
    console.log(id);
  };

  const styles = {
    card: {
      width: "200px",
      height: "300px",
      position: "relative",
      bgcolor: "secondary.dark",
      ":hover": {
        transition: "0.6s ease",
        opacity: 0.8,
      },
      ":hover .rating": {
        display: "none",
      },
    },
    hoverLayer: {
      width: "inherit",
      height: "220px",
      position: "absolute",
      bgcolor: "secondary.dark",
      opacity: 0,
      textAlign: "center",
      pt: "60%",
      cursor: "pointer",
      ":hover": {
        transition: " .3s ease",
        opacity: 1,
      },
    },
    playIcon: { color: "primary.light", fontSize: "60px" },
    ratingBox: {
      position: "absolute",
      top: "20px",
      left: "0px",
      borderRadius: "5px",
      height: "25px",
      width: "40px",
      bgcolor: "green",
      textAlign: "center",
    },
    cardImage: {
      width: "180px",
      ml: 1,
      mt: 1,
      borderRadius: "10px",
    },
    genres: { display: "flex", textAlign: "center", columnGap: 1 },
  };

  return (
    <Card sx={styles.card}>
      <Box sx={styles.hoverLayer} onClick={handleClick}>
        <PlayCircle sx={styles.playIcon} />
      </Box>

      <Box className="rating" sx={styles.ratingBox}>
        <Typography variant="subtitle1" component="span">
          {vote_average}
        </Typography>
      </Box>
      <CardMedia
        component="img"
        height="220px"
        image={poster_path}
        sx={styles.cardImage}
      />
      <CardContent sx={{ p: 1 }}>
        <Typography variant="h6" color="primary.light">
          {title}
        </Typography>
        <Typography variant="subtitle2" sx={styles.genres} color="primary.dark">
          {genres}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default MovieCard;
