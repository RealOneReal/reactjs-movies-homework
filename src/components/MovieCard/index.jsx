import { Card, Typography, CardMedia, CardContent, Box } from "@mui/material";
import { PlayCircle } from "@mui/icons-material";
import { truncateString } from "../../utils/stringUtils";

const MovieCard = (props) => {
  const { poster_path, genre_ids, vote_average, original_title, id } = props;
  const title = truncateString(original_title, 19);
  const genres = truncateString(genre_ids?.length && genre_ids.join(" "), 26);

  const handleClick = () => {
    console.log(id);
  };

  const styles = {
    container: {
      position: "relative",
      ":hover .MuiPaper-root": {
        transition: "0.6s ease",
        opacity: 0.3,
      },
    },
    card: {
      width: "200px",
      height: "300px",
      position: "relative",
      bgcolor: "secondary.dark",
    },
    hoverLayer: {
      width: "200px",
      height: "300px",
      position: "absolute",
      left: 0,
      top: 0,
      opacity: 0,
      textAlign: "center",
      cursor: "pointer",
      ":hover": {
        transition: " .3s ease",
        opacity: 1,
      },
    },
    playIcon: {
      color: "primary.main",
      fontSize: "60px",
      position: "absolute",
      top: "calc(50% - 30px)",
      left: "calc(50% - 30px)",
    },
    ratingBox: {
      position: "absolute",
      top: "20px",
      left: "0px",
      borderRadius: "5px",
      height: "25px",
      width: "40px",
      bgcolor: "green",
      textAlign: "center",
    },
    cardImage: {
      width: "180px",
      ml: 1,
      mt: 1,
      borderRadius: "10px",
    },
    genres: { display: "flex", textAlign: "center", columnGap: 1 },
  };

  return (
    <Box sx={styles.container}>
      <Card sx={styles.card}>
        <Box className="rating" sx={styles.ratingBox}>
          <Typography variant="subtitle1" component="span">
            {vote_average}
          </Typography>
        </Box>
        <CardMedia
          component="img"
          height="220px"
          image={poster_path}
          sx={styles.cardImage}
        />
        <CardContent sx={{ p: 1 }}>
          <Typography variant="h6" color="primary.light">
            {title}
          </Typography>
          <Typography
            variant="subtitle2"
            sx={styles.genres}
            color="primary.dark"
          >
            {genres}
          </Typography>
        </CardContent>
      </Card>
      <Box sx={styles.hoverLayer} onClick={handleClick}>
        <PlayCircle sx={styles.playIcon} />
      </Box>
    </Box>
  );
};

export default MovieCard;
