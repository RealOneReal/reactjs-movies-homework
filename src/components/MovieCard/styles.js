const styles = {
  container: {
    position: "relative",
    ":hover .MuiPaper-root": {
      transition: "0.6s ease",
      opacity: 0.3,
    },
  },
  card: {
    width: "200px",
    height: "300px",
    position: "relative",
    bgcolor: "secondary.dark",
  },
  hoverLayer: {
    width: "200px",
    height: "300px",
    position: "absolute",
    left: 0,
    top: 0,
    opacity: 0,
    textAlign: "center",
    cursor: "pointer",
    ":hover": {
      transition: " .3s ease",
      opacity: 1,
    },
  },
  playIcon: {
    color: "primary.main",
    fontSize: "60px",
    position: "absolute",
    top: "calc(50% - 30px)",
    left: "calc(50% - 30px)",
  },
  ratingBox: {
    position: "absolute",
    top: "20px",
    left: "0px",
    borderRadius: "5px",
    height: "25px",
    width: "40px",
    bgcolor: "green",
    textAlign: "center",
  },
  cardImage: {
    width: "180px",
    height: "220px",
    ml: 1,
    mt: 1,
    borderRadius: "10px",
  },
  genres: { display: "flex", textAlign: "center", columnGap: 1 },
};

export default styles;
