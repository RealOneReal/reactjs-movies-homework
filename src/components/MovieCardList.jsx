import { Grid } from "@mui/material";
import MovieCard from "./MovieCard";

const MovieCardList = (props) => {
  const { movieCards } = props;
  return (
    <Grid container>
      {movieCards.map((card) => (
        <Grid item xs={9} md={2} key={card.id}>
          <MovieCard props={card} />
        </Grid>
      ))}
    </Grid>
  );
};
export default MovieCardList;
