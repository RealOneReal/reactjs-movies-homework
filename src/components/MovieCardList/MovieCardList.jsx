import { Grid } from "@mui/material";
import MovieCard from "../MovieCard/MovieCard";

const MovieCardList = (props) => {
  const { results } = props;
  return (
    <Grid container spacing={1} justifyContent="center">
      {results.map((card) => (
        <Grid item key={card.id}>
          <MovieCard {...card} />
        </Grid>
      ))}
    </Grid>
  );
};
export default MovieCardList;
