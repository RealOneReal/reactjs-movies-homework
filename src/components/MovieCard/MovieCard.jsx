import { Card, Typography, CardMedia, CardContent, Box } from "@mui/material";
import { PlayCircle } from "@mui/icons-material";
import fallback from "../../assets/broken-image.png";
import styles from "./styles";
import { truncateString } from "../../utils/stringUtils";

const MovieCard = ({ posterPath, genreIds, voteAverage, title, id }) => {
  const movieTitle = truncateString(title, 19);
  const genres = truncateString(genreIds.join(" "), 26);

  const handleClick = () => {
    console.log(id);
  };

  const handleMediaFallback = ({ target }) => {
    target.style.objectFit = "contain";
    return (target.src = fallback);
  };

  return (
    <Box sx={styles.container} onClick={handleClick}>
      <Card sx={styles.card}>
        <Box className="rating" sx={styles.ratingBox}>
          <Typography variant="subtitle1" component="span">
            {voteAverage}
          </Typography>
        </Box>
        <CardMedia
          component="img"
          alt={movieTitle}
          src={posterPath}
          sx={styles.cardImage}
          onError={handleMediaFallback}
        />
        <CardContent sx={{ p: 1 }}>
          <Typography variant="h6" color="primary.light">
            {movieTitle}
          </Typography>
          <Typography
            variant="subtitle2"
            sx={styles.genres}
            color="primary.dark"
          >
            {genres}
          </Typography>
        </CardContent>
      </Card>
      <Box sx={styles.hoverLayer}>
        <PlayCircle sx={styles.playIcon} />
      </Box>
    </Box>
  );
};

export default MovieCard;
