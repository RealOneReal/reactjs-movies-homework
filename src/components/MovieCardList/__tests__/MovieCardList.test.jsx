import renderer from "react-test-renderer";
import MovieCard from "../../MovieCard/MovieCard";
import MovieCardList from "../MovieCardList";

const mockData = {
  results: [
    {
      original_title: "Brat 1",
      id: 1,
    },
    {
      original_title: "Brat 2",
      id: 2,
    },
    {
      original_title: "Brat 3",
      id: 3,
    },
  ],
};

jest.mock("../../MovieCard/MovieCard", () => {
  return "MovieCard";
});

describe("MovieCardList", () => {
  it("should be render", () => {
    const tree = renderer.create(<MovieCardList {...mockData} />);
    expect(tree).toMatchSnapshot();
  });
});
