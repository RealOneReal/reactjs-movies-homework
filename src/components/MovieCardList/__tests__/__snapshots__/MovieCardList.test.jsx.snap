// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardList should be render 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-exfl4s-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <MovieCard
      id={1}
      original_title="Brat 1"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <MovieCard
      id={2}
      original_title="Brat 2"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <MovieCard
      id={3}
      original_title="Brat 3"
    />
  </div>
</div>
`;
